numba/targets/arrayobj.py

```diff
@@ -1838,6 +1838,31 @@ def np_unique_impl(a):
     return np_unique_impl
 
 
+@overload(np.repeat)
+def np_repeat(a, repeats):
```

Comment:
The typing part of this function should really ensure that `a` is of a an array-like type (in the current implementation, only `numba.types.Array` will work), and that `repeats` is of `numba.types.Integer` integer type.

Reply:
Hey, can you explain this a little more. I am not really sure if I got it.

Reply:
Sure, below I've annotated a `@overload` use as a template to follow. Hope this helps.

```python
# Declare that function `FOO` is going to be overloaded (have a substitutable
# numba implementation)
@overload(FOO)

# Define the overload function with formal arguments
# these arguments must be matched in the inner function implementation
def JIT_FOO(arg0, arg1, arg2, ...):

    # This scope is for typing, access is available to the *type* of `a` and
    # `repeats`, for example `a` might be a `numba.types.array(int64, 1, 'C')`
    # and `repeats` in `int64`. This information can be used to change the
    # behaviour of the implementing function and check that the types are
    # actually supported by the implementation.
    print(arg0) # this will show the Numba type system type of arg0

    # This is the definition of the function that implements the `FOO` work. It
    # does whatever algorithm is need to implement FOO.
    def FOO_impl(arg0, arg1, arg2, ...): # match arguments to JIT_FOO above
        # < Implementation of FOO goes here >
        return # whatever needs to be returned by the FOO algorithm

    # return the implementation
    return FOO_impl
```

Reply:
Hey, I read through and tried this again. By this you mean I should just be printing the type of the argument? Not modifying it?

Reply:
The print statement was there for pedagogical reasons, it's to encourage you to print out or inspect the variables in that scope so that you can see what's going on. The types themselves don't need modifying, they need checking to ensure that they are of a type that the implementation `FOO_impl` expects.